sky/shell/platform/mac/platform_mac.mm

```diff
@@ -83,17 +88,23 @@ static void RedirectIOConnectionsToSyslog() {
     sky::shell::Shell::InitStandalone(icu_data_path);
   }
 
+  ~EmbedderState() {
+    CHECK([NSThread isMainThread])
+        << "Embedder destruction must occur on the main platform thread";
+#if !TARGET_OS_IPHONE
+    embedder_message_loop_->QuitNow();
+    embedder_message_loop_.release();
```

Comment:
We leak the instance. Though there is a `MessagePumpMac::Create()`, there is no corresponding way to collect that instance. And there is an assertion in the destructor of the message loop that trips on that check.